src/Core/Scheduler/Providers/$3dTiles_Provider.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Created On: 2016-11-8
+ * Class: $3dTiles_Provider
+ * Description:
+ */
+import * as THREE from 'three';
+import Provider from './Provider';
+import B3dmLoader from '../../../Renderer/ThreeExtended/B3dmLoader';
+import Fetcher from './Fetcher';
+import BasicMaterial from '../../../Renderer/BasicMaterial';
+
+
+export function $3dTilesIndex(tileset, urlPrefix) {
+    let counter = 0;
+    this.index = {};
+    const recurse = function recurse_f(node, urlPrefix) {
+        this.index[counter] = node;
+        node.tileId = counter;
+        node.urlPrefix = urlPrefix;
+        counter++;
+        if (node.children) {
+            for (const child of node.children) {
+                recurse(child, urlPrefix);
+            }
+        }
+    }.bind(this);
+    recurse(tileset.root, urlPrefix);
+
+    this.extendTileset = function extendTileset(tileset, nodeId, urlPrefix) {
+        recurse(tileset.root, urlPrefix);
+        this.index[nodeId].children = [tileset.root];
+    };
+}
+
+function $3dTiles_Provider(/* options*/) {
+    // Constructor
+
+    Provider.call(this);
+    this.b3dmLoader = new B3dmLoader();
+}
+
+$3dTiles_Provider.prototype = Object.create(Provider.prototype);
+
+$3dTiles_Provider.prototype.constructor = $3dTiles_Provider;
+
+$3dTiles_Provider.prototype.removeLayer = function removeLayer(/* idLayer*/) {
+
+};
+
+$3dTiles_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(/* layer*/) {
+
+};
+
+function getBox(boundingVolume) {
+    if (boundingVolume.region) {
+        // TODO: unsupported
+        return { region: boundingVolume.region };
+    } else if (boundingVolume.box) {
+        // TODO: only works for axis aligned boxes
+        const box = boundingVolume.box;
+        const center = new THREE.Vector3(box[0], box[1], box[2]);
+        const w = center.x - box[3];
+        const e = center.x + box[3];
+        const s = center.y - box[7];
+        const n = center.y + box[7];
+        const b = center.z - box[11];
+        const t = center.z + box[11];
+
+        return { box: new THREE.Box3(new THREE.Vector3(w, s, b), new THREE.Vector3(e, n, t)) };
+    } else if (boundingVolume.sphere) {
+        // TODO
+        return { sphere: undefined };
```

Comment:
Maybe all the unimplemented code path should `throw new Error()..)` instead of silently failing?